### PR TITLE
🍎 Prepare for "Build JVips dependencies on MacOS"?

### PR DESCRIPTION
**Pull request by anthonyroussel**
Hello

The goal of this pull-request to:
    
* Harmonise the build between the different architectures (linux, w64, macOS)
* Install only the build dependencies required to build Jvips
* Speed up the build of JVips on MacOS
* Trigger the MacOS build when a pull-request is created
* Upgrade libexif to 0.6.24

However, this has its drawbacks:
    
* `lib/CMakeLists.txt` has to be maintained for MacOS

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -56,7 +56,6 @@ jobs:
 
   build_for_macos:
     runs-on: macos-latest
-    if: github.event_name != 'pull_request' # To speed up the feedback
```
Comment (dbouron) [inline .github/workflows/build.yml:59]: In the past, the macos build took more than 30 min this is why we remove this jobs from PR event.
Last build took 16 min (~10 min for Linux) which is acceptable. So we can enable it back.

File: lib/VERSIONS [hunk 2]
```diff
@@ -2,7 +2,7 @@
 LIQ_VERSION=2.12.6
 
 # renovate: datasource=github-releases depName=libexif packageName=libexif/libexif
-EXIF_VERSION=0.6.22
+EXIF_VERSION=0.6.24
```
Comment (dbouron) [inline lib/VERSIONS:5]: lgtm

File: lib/CMakeLists.txt [hunk 3]
```diff
@@ -19,11 +19,13 @@ if (NOT DEFINED BUILD_TARGET)
...
   set(CONFIGURE_HOST "--host=x86_64-w64-mingw32")
+elseif(${BUILD_TARGET} STREQUAL "macOS")
+  set(CONFIGURE_HOST "--host=x86_64-apple-darwin")
 endif()
 
 if (NOT DEFINED CMAKE_BUILD_TYPE)
-set(CONFIGURE_CFLAGS "${CMAKE_C_FLAGS}")
-set(CONFIGURE_CXXFLAGS "${CMAKE_CXX_FLAGS}")
+  set(CONFIGURE_CFLAGS "${CMAKE_C_FLAGS}")
+  set(CONFIGURE_CXXFLAGS "${CMAKE_CXX_FLAGS}")
```
Comment (dbouron) [inline lib/CMakeLists.txt:28]: Good catch :wrench:

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -69,7 +69,7 @@ jobs:
           java-package: jdk
           architecture: x64
 
-      - name: Install vips
+      - name: Install build dependencies
```
Comment (dbouron) [inline .github/workflows/build.yml:71]: I am pretty sure that it will be the root of all evil :apple:

File: lib/CMakeLists.txt [hunk 5]
```diff
@@ -237,6 +237,9 @@ if (WITH_LIBHEIF)
           -DCMAKE_INSTALL_LIBDIR=lib
           -DBUILD_SHARED_LIBS=1
           -DENABLE_CCACHE=0
+          -DENABLE_TESTS=0
+          -DENABLE_EXAMPLES=0
+          -DENABLE_DOCS=0
```
Comment (dbouron) [inline lib/CMakeLists.txt:242]: Thank you for this contribution :slightly_smiling_face:

File: Toolchain-macOS.cmake [hunk 6]
```diff
@@ -2,6 +2,7 @@ set(BUILD_TARGET macOS)
 
 # the name of the target operating system
 SET(CMAKE_SYSTEM_NAME Darwin)
+set(CMAKE_SYSTEM_PROCESSOR AMD64)
```
Comment (dbouron) [inline Toolchain-macOS.cmake:5]: NIT: `set` in uppercase for style homogeneity please

**Review by dbouron — CHANGES_REQUESTED** [review]
Hi @anthonyroussel,

Thank you for this bunch of commits.

We can merge all commits except the one for building macos deps. I'd like to merge some [others dependencies](https://github.com/criteo/JVips/pull/125/commits) first.
Can you please split your PR?

> `lib/CMakeLists.txt` has to be maintained for MacOS

**Comment by anthonyroussel** [comment]
> Hi @anthonyroussel,
> 
> Thank you for this bunch of commits.
> 
> We can merge all commits except the one for building macos deps. I'd like to merge some [others dependencies](https://github.com/criteo/JVips/pull/125/commits) first. Can you please split your PR?
> 
> > `lib/CMakeLists.txt` has to be maintained for MacOS
> 
> 

Hello,

Thanks for the review!

No problem, I removed the "Build JVips dependencies on MacOS" commit from this PR :)

**Review by dbouron — APPROVED** [review]

**Comment by dbouron** [comment]
Hi @anthonyroussel,
Thank you for your previous changes. I rebased your other PR which produced some conflicts. Sorry about that.

**Comment by anthonyroussel** [comment]
> Hi @anthonyroussel, Thank you for your previous changes. I rebased your other PR which produced some conflicts. Sorry about that.

Hello

No problem, I just resolved the conflicts :)

**Pull request merged**